### ISSUE TITLE
autocommit for TConnectionPool throws exception

### ISSUE THREAD

**Issue by bosswissam** [eventlet/eventlet#389]
Hi Team

I have a simple setup, attempting to create a pool of connections and set each of them to autocommit mode, however that is not possible without accessing the hidden `._base` object on the returned connection (wrapper):

```python
import psycopg2
from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
from eventlet.db_pool import ConnectionPool
import eventlet
c = ConnectionPool(psycopg2, host='xxx', user='user', database='db', password='pwd', min_size=1, max_size=2)
conn = c.get()
conn.autocommit # <- this is defined as a property on http://bit.ly/2lRSfwn, but defined as a function here
conn.autocommit # <- this throws an exception because it attempts to call .autocommit on the base psycopg2.connection object, which has this defined as a property not a function
conn._base.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT) #<- works
```

It doesn't feel right to access the `_base` property - are you planning to expose an interface to set autocommit without having to do so? Or did I miss something in your docs? Finally, the currently autocommit as a function in the GenericConnectionWrapper wrapper object will always raise an exception.

**Comment by temoto** [comment]
Please, try this patch.

`pip install https://github.com/eventlet/eventlet/archive/dbpool-389.zip`

**Comment by temoto** [comment]
The fix is merged into master 885482b57691bd19e8b403e43a035546705da21e

I verified `conn.set_isolation_level(...)` now works without accessing `conn._base`.

Be aware that connection will be returned to pool "as is", without resetting isolation level. To avoid that, simplest and most robust solution is to close connection before returning it back to pool.
```
with pool.item() as conn:
  conn.set_isolation_level(...)
  # use connection as you like
  conn.close()
```

Also for psycopg2 and all pure python DB drivers I recommend using `eventlet.db_pool.RawConnectionPool`, without OS threads. Unless you have a better reason.
